components/ActionSheet/src/private/MDCActionSheetHeaderView.m

```diff
@@ -0,0 +1,258 @@
+  }
+#endif
+  return newWidth;
+}
+
+- (void)setTitle:(NSString *)title {
+  titleLabel.text = title;
+  [self setNeedsLayout];
+}
+
+- (NSString *)title {
+  return titleLabel.text;
+}
+
+- (void)setMessage:(NSString *)message {
+  messageLabel.text = message;
+  [self setNeedsLayout];
+}
+
+- (NSString *)message {
+  return messageLabel.text;
+}
+
+- (void)setTitleFont:(UIFont *)titleFont {
+  _titleFont = titleFont;
+  [self updateTitleFont];
+}
+
+- (void)setMessageFont:(UIFont *)messageFont {
+  _messageFont = messageFont;
+  [self updateMessageFont];
+}
+
++ (UIFont *)titleFontDefault {
+  if ([MDCTypography.fontLoader isKindOfClass:[MDCSystemFontLoader class]]) {
+    return [UIFont mdc_standardFontForMaterialTextStyle:MDCFontTextStyleSubheadline];
+  }
+  return [MDCTypography subheadFont];
+}
+
++ (UIFont *)messageFontDefault {
+  if ([MDCTypography.fontLoader isKindOfClass:[MDCSystemFontLoader class]]) {
+    return [UIFont mdc_standardFontForMaterialTextStyle:MDCFontTextStyleBody1];
+  }
+  return [MDCTypography body1Font];
+}
+
+- (void)updateFonts {
+  [self updateTitleFont];
+  [self updateMessageFont];
+}
+
+- (void)updateTitleFont {
+  UIFont *titleFont = _titleFont ?: [[self class] titleFontDefault];
+  if (self.mdc_adjustsFontForContentSizeCategory) {
+    titleLabel.font =
+    [titleFont mdc_fontSizedForMaterialTextStyle:MDCFontTextStyleSubheadline
+                            scaledForDynamicType:self.mdc_adjustsFontForContentSizeCategory];
+  } else {
+    titleLabel.font = titleFont;
+  }
+  [self setNeedsLayout];
+}
+
+- (void)updateMessageFont {
+  UIFont *messageFont = _messageFont ?: [[self class] messageFontDefault];
+  if (self.mdc_adjustsFontForContentSizeCategory) {
+    messageLabel.font =
+    [messageFont mdc_fontSizedForMaterialTextStyle:MDCFontTextStyleBody1
+                              scaledForDynamicType:self.mdc_adjustsFontForContentSizeCategory];
+  } else {
+    messageLabel.font = messageFont;
+  }
+
+  [self setNeedsLayout];
+}
+
+- (void)mdc_setAdjustsFontForContentSizeCategory:(BOOL)adjusts {
+  mdc_adjustsFontForContentSizeCategory = adjusts;
+  [self updateFonts];
+}
+
+- (CGSize)sizeThatFits:(CGSize)size {
+  CGSize boundsSize = CGRectInfinite.size;
+  boundsSize.width = [self accomodateSafeAreaInWidth:size.width];
+
+  CGSize contentSize = [self calculateContentSizeThatFitsWidth:boundsSize.width];
+  return contentSize;
+}
+
+- (CGSize)calculateContentSizeThatFitsWidth:(CGFloat)boundingWidth {
+  CGSize boundsSize = CGRectInfinite.size;
+  boundsSize.width = boundingWidth - kLeadingPadding - kTrailingPadding;
+
+  CGSize titleSize = [titleLabel sizeThatFits:boundsSize];
+  CGSize messageSize = [messageLabel sizeThatFits:boundsSize];
+
+  CGFloat contentWidth = MAX(titleSize.width, messageSize.width);
+
+
+  contentWidth = contentWidth + kLeadingPadding + kTrailingPadding;
+
+  CGFloat contentHeight;
+  BOOL messageCheck = (self.message == nil) || ([self.message isEqualToString:@""]);
```

Comment:
naming is hard.
Consider messageEmpty?
And it's sometimes easier to deal with positives, so consider inverting this logic and calling it messageExists.
Alternative, it not uncommon to see in code:
If (self.message && !self.message.isEmpty)